test/qdsht.jl

```diff
@@ -0,0 +1,322 @@
+        @test size(AAs, 1) == 2 * length(A) + 1
+        for i in 1:2
+            @test AAs[1:128, i] == A[128:-1:1]
+            @test AAs[129, i] ≈ 1
+            @test AAs[130:end, i] == A
+        end
+    end
+
+    @testset "multiple orders/spherical dimensions" begin
+        @testset "p=$p, n=$n" for p in (0, 1 / 2, 1, 2, 3), n in (1, 2, 3)
+            #= Test case from Guizar-Sicairos and Gutierrez-Vega,
+              "Computation of quasi-discrete Hankel transforms of integer order for propagating
+              optical wave fields" =#
+            # Adapted from Gradshteyn and Ryzhik 6.671.1
+            γ = 5
+            f(r) = r^(-(n + 1) / 2) * sin(2π * γ * r)
+            function fk(k)
+                s = p + (n - 1) / 2
+                return if k < 2π * γ
+                    return k^s * cos(s * π / 2) / sqrt(4 * π^2 * γ^2 - k^2) /
+                           (2π * γ + sqrt(4 * π^2 * γ^2 - k^2))^s / k^((n - 1) / 2)
+                else
+                    return sin(s * asin(2π * γ / k)) / sqrt(k^2 - 4 * π^2 * γ^2) /
+                           k^((n - 1) / 2)
+                end
+            end
+
+            R = 1
+            N = 256
+            q1 = QDSHT(p, n, R, N)
+            v = f.(q1.r)
+            vk = q1 * v
+            vka = fk.(q1.k)
+            err = dynε(vka, vk)
+            i = findlast(ki -> ki < 2π * γ, q1.k) # find point flanking discontinuity
+            @test maximum(dynε(vka, vk)[[1:(i - 1); (i + 2):end]]) < -10
+            if p != 0
+                @test_throws DomainError onaxis(vk, q1)
+            end
+            @test integrateR(abs2.(v), q1) ≈ integrateK(abs2.(vk), q1)
+            # curve is too steep for quadrature, so compare with integral computed with
+            # Mathematica
+            @test isapprox(integrateR(abs2.(v), q1), 2.358965372, rtol = 1e-2)
+
+            R = 4e-2
+            N = 256
+            w0 = 1e-2
+            a = 2 / w0
+            q4 = QDSHT(p, n, R, N)
+            g(r) = exp(-1 // 2 * a^2 * r^2)
+            v = g.(q4.r)
+            vk = q4 * v
+            @test integrateR(abs2.(v), q4) ≈ integrateK(abs2.(vk), q4)
+            # note the high tolerance here is due to the lack of samples close to the axis
+            @test isapprox(
+                integrateR(abs2.(v), q4),
+                quad_integrate(q4, abs2 ∘ g),
+                rtol = 1e-2,
+            )
+
+            @testset "pdf non-central chi" begin
+                R = 30
+                N = 256
+                q5 = Hankel.QDSHT(p, n, R, N)
+                h(r) = pdf_ncchi(r, 10, 1, n)
+                hk(k) = trans_pdf_ncchi(k, 10, 1, n, p)
+
+                v = h.(q5.r)
+                vk = q5 * v
+                vka = hk.(q5.k)
+                @test maximum(dynε(vka, vk)) < -10
+                @test integrateR(abs2.(v), q5) ≈ integrateK(abs2.(vk), q5)
+            end
+        end
+    end
+
+    @testset "QDSHT === QDHT when n = 1" begin
```

Comment:
here you're not testing `onaxis` even for p=0 -- is there a reason for that?

Reply:
No, I can add a test